File: wrench-app/src/main/java/com/izettle/wrench/database/WrenchConfigurationValueDao.java
package com.izettle.wrench.database;

import com.izettle.wrench.database.tables.ConfigurationValueTable;

import java.util.List;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

@Dao
public interface WrenchConfigurationValueDao {

    @Query("SELECT * FROM " + ConfigurationValueTable.TABLE_NAME +
            " WHERE " + ConfigurationValueTable.COL_CONFIG_ID + " = (:configurationId)")
    LiveData<List<WrenchConfigurationValue>> getConfigurationValue(long configurationId);

    @Query("SELECT * FROM " + ConfigurationValueTable.TABLE_NAME +
            " WHERE " + ConfigurationValueTable.COL_CONFIG_ID + " = (:configurationId) AND " + ConfigurationValueTable.COL_SCOPE + " = (:scopeId)")
    LiveData<WrenchConfigurationValue> getConfigurationValue(long configurationId, long scopeId);

    @Query("UPDATE " + ConfigurationValueTable.TABLE_NAME +
            " SET " + ConfigurationValueTable.COL_VALUE + " = (:value)" +
            " WHERE " + ConfigurationValueTable.COL_CONFIG_ID + " = (:configurationId) AND " + ConfigurationValueTable.COL_SCOPE + " = (:scopeId) ")
    int updateConfigurationValue(long configurationId, long scopeId, String value);

    @Insert
    long insert(WrenchConfigurationValue wrenchConfigurationValue);

    @Update
    int update(WrenchConfigurationValue wrenchConfigurationValue);

    @Delete
    void delete(WrenchConfigurationValue selectedConfigurationValue);
}

File: wrench-app/src/main/java/com/izettle/wrench/database/WrenchConfigurationValueDao.kt
package com.izettle.wrench.database

import androidx.lifecycle.LiveData
import androidx.room.*
import com.izettle.wrench.database.tables.ConfigurationValueTable

@Dao
abstract class WrenchConfigurationValueDao {

    @Query("SELECT * FROM " + ConfigurationValueTable.TABLE_NAME +
            " WHERE " + ConfigurationValueTable.COL_CONFIG_ID + " = (:configurationId)")
    abstract fun getConfigurationValue(configurationId: Long): LiveData<List<WrenchConfigurationValue>>

    @Query("SELECT * FROM " + ConfigurationValueTable.TABLE_NAME +
            " WHERE " + ConfigurationValueTable.COL_CONFIG_ID + " = (:configurationId) AND " + ConfigurationValueTable.COL_SCOPE + " = (:scopeId)")
    abstract fun getConfigurationValue(configurationId: Long, scopeId: Long): LiveData<WrenchConfigurationValue>

    @Query("UPDATE " + ConfigurationValueTable.TABLE_NAME +
            " SET " + ConfigurationValueTable.COL_VALUE + " = (:value)" +
            " WHERE " + ConfigurationValueTable.COL_CONFIG_ID + " = (:configurationId) AND " + ConfigurationValueTable.COL_SCOPE + " = (:scopeId) ")
    abstract fun updateConfigurationValueSync(configurationId: Long, scopeId: Long, value: String): Int

    @Query("UPDATE " + ConfigurationValueTable.TABLE_NAME +
            " SET " + ConfigurationValueTable.COL_VALUE + " = (:value)" +
            " WHERE " + ConfigurationValueTable.COL_CONFIG_ID + " = (:configurationId) AND " + ConfigurationValueTable.COL_SCOPE + " = (:scopeId) ")
    abstract suspend fun updateConfigurationValue(configurationId: Long, scopeId: Long, value: String): Int

    @Insert
    abstract fun insertSync(wrenchConfigurationValue: WrenchConfigurationValue): Long

    @Insert
    abstract suspend fun insert(wrenchConfigurationValue: WrenchConfigurationValue): Long

    @Update
    abstract fun update(wrenchConfigurationValue: WrenchConfigurationValue): Int

    @Delete
    abstract suspend fun delete(selectedConfigurationValue: WrenchConfigurationValue)
}
